jobs/base_model_validator.py

```diff
@@ -66,6 +67,29 @@ def process(self, input_model, regex_string):
             yield errors.ModelInvalidIdError(model)
 
 
+class ValidatePostCommitIsPrivate(beam.DoFn):
+    """DoFn to check if post_commmit_status is private when
+    post_commit_is_private is true and vice-versa."""
+
+    def process(self, input_model):
+        """Function validates that post_commit_is_private is true iff
+        post_commit_status is private
+
+        Args:
+            input_model: datastore_services.Model. Entity to validate.
+
+        Yields:
+            ModelInvalidCommitStatus. Error for commit_type validation.
+        """
+        model = jobs_utils.clone_model(input_model)
+        if model.post_commit_status == feconf.POST_COMMIT_STATUS_PRIVATE and (
+                not model.post_commit_is_private):
+            yield errors.ModelInvalidCommitStatusError(model)
+        if model.post_commit_status == feconf.POST_COMMIT_STATUS_PUBLIC and (
+                model.post_commit_is_private):
+            yield errors.ModelInvalidCommitStatusError(model)
```

Comment:
I think this may be simpler:

```python
        expected_post_commit_is_private = (
            model.post_commit_status == feconf.POST_COMMIT_STATUS_PRIVATE)
        if model.post_commit_is_private != expected_post_commit_is_private:
            yield errors.ModelInvalidCommitStatusError(model)
```

Reply:
Done